docs/src/pages/components/data-grid/group-pivot/group-pivot.md

```diff
@@ -6,6 +6,197 @@ title: Data Grid - Group & Pivot
+The callback is called for each grouping column, and it receives the respective column's "fields" as parameter.
+
+{{"demo": "pages/components/data-grid/group-pivot/RowGroupingCustomGroupingColDefCallback.js", "bg": "inline", "defaultCodeOpen": false}}
+
+#### Show values for the leaves
+
+By default, the grouped rows display no value on their grouping columns' cells. We're calling those cells "leaves".
+
+If you want to display some value, you can provide a `leafField` property to the `groupingColDef`.
+
+{{"demo": "pages/components/data-grid/group-pivot/RowGroupingLeafWithValue.js", "bg": "inline", "defaultCodeOpen": false}}
+
+#### Hide the descendant count
+
+Use the `hideDescendantCount` property of the `groupingColDef to hide the number of descendants of a grouping row.
+
+{{"demo": "pages/components/data-grid/group-pivot/RowGroupingHideDescendantCount.js", "bg": "inline", "defaultCodeOpen": false}}
+
+### Disable the row grouping
+
+#### For all columns
+
+You can disable row grouping by setting `disableRowGrouping` prop to true.
+
+It will disable all the features related to the row grouping, even if a model is provided.
+
+{{"demo": "pages/components/data-grid/group-pivot/RowGroupingDisabled.js", "bg": "inline", "defaultCodeOpen": false}}
+
+#### For some columns
+
+In case you need to disable grouping on specific column(s), set the `groupable` property on the respective column definition (`GridColDef`) to `false`.
+In the example below, the `director` column can not be grouped. And in all example, the `title` and `gross` columns can not be grouped.
+
+{{"demo": "pages/components/data-grid/group-pivot/RowGroupingColDefCanBeGrouped.js", "bg": "inline", "defaultCodeOpen": false}}
+
+### Using 'keyGetter' for complex grouping value
+
+In most scenarios, to handle complex values (e.g. nested structures), a `valueGetter` property must be provided to the column definition to convert it into a simple value.
+But sometimes, you may want to keep the value in its raw format, to be used in a custom cell renderer, and thus using a value getter is not possible.
+In this case, pass a `keyGetter` property to the column definition to convert this object into a serializable value.
+
+```ts
+const columns: GridColumns = [
+  {
+    field: 'composer',
+    keyGetter: (params) => params.value.name,
+  },
+  // ...
+];
+```
+
+{{"demo": "pages/components/data-grid/group-pivot/RowGroupingKeyGetter.js", "bg": "inline", "defaultCodeOpen": false}}
+
+**Note**: If your column also have a `valueGetter` property, its value is used as the parameter for the `keyGetter` method.
+
+{{"demo": "pages/components/data-grid/group-pivot/RowGroupingKeyGetterValueGetter.js", "bg": "inline", "defaultCodeOpen": false}}
+
+### Rows with missing groups
+
+If the grouping key of a grouping criteria is `null` or `undefined` for a row, the grid will consider that this row does not have a value for this group. and will inline it for those groups.
```

Comment:
We might need to change this behavior for #213. I see rows with `null` or `undefined` as another group, often times called "N.A.", "Empty", "Other", etc.

DevExpress groups them not matter the value. See https://codesandbox.io/s/gracious-ride-fo79o?file=/demo-data/generator.js

Reply:
https://www.ag-grid.com/javascript-data-grid/grouping-unbalanced-groups/

AG-Grid does not and suggest to use an arbitrary value to represent this groupable nullish values.


What would be blocking for #213 ?

Reply:
> What would be blocking for #213 ?

I suppose that the aggregated values will be displayed in the group row or at the footer of the current group. If these rows don't belong to a group, there's no place to display the aggregated values.

Reply:
Not sure to follow you here.
My point was just that AG-Grid consider `null` and `undefined` as non-groupable values. Which enable the creation of unbalanced group (cf my link). In the Tree Structure, they are still in a group, the group with "no value" for this criteria.

For instance if you group by year and director and some movies don't have a director.
Then the movies with a director are in the groups:
- `[{ groupingField: 'year', groupingValue: 2020 }, { groupingField: 'director', groupingValue: 'A' }]`
- `[{ groupingField: 'year', groupingValue: 2020 }, { groupingField: 'director', groupingValue: 'B' }]`
- ...
- 
And the movies without a director are in the groups:
- `[{ groupingField: 'year', groupingValue: 2020 }]
- `[{ groupingField: 'year', groupingValue: 2019 }]
- ...

And for Aggregation, for me we could store the aggregated values on the node of the parent
Which would make us create a parent node for the top level rows (which don't exist right now but could easily be added).

As for where those values should be rendered, as you said AG-Grid has 2 places:
- The footer: https://www.ag-grid.com/javascript-data-grid/aggregation/, which looks like a row below the last row of the group
- The grouping row: https://www.ag-grid.com/javascript-data-grid/aggregation/

But as every row is in a group, I don't think unbalanced groups are an issue here.

Reply:
In the link you added it doesn't group rows with `null` or `undefined`. In the "sum(Val1)" column for these rows there's the raw values instead. Anyway, it's a small detail that I would expect that grid would handle automatically, but we can keep the same behavior as AG Grid.